scene/gui/color_picker.cpp

```diff
@@ -2442,12 +2442,14 @@ void ColorPickerButton::_color_changed(const Color &p_color) {
 }
 
 void ColorPickerButton::_modal_closed() {
-	if (Input::get_singleton()->is_action_just_pressed(SNAME("ui_cancel"))) {
-		set_pick_color(picker->get_old_color());
-		emit_signal(SNAME("color_changed"), color);
+	if (picker->is_visible_in_tree()) {
```

Comment:
This can be early return:

```suggestion
	if (!picker->is_visible_in_tree()) {
		return;
	}
```

Reply:
I considered this but didn't want to interfere with the previous fix, I'll test it now though and see. In theory it shouldn't make a difference.

Reply:
Unfortunately an early return **does** break the previous fix.